esmvalcore/preprocessor/_time.py

```diff
@@ -711,3 +754,73 @@ def timeseries_filter(cube, window, span,
                                weights=wgts)
 
     return cube
+
+
+def resample_hours(cube, hours):
+    """
+    Converts x-hourly data to y-hourly by eliminating extra timesteps.
+
+    Converts x-hourly data to y-hourly (y > x) by eliminating the extra
+    timesteps. This is intended to be used only with instantaneous values.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        Input cube.
+    hours: int
+        The frequency of the desired data.
+
+    Returns
+    -------
+    iris.cube.Cube
+        Cube with the new frequency.
+
+    Raises
+    ------
+    ValueError:
+        The specified freuqncy is not a divisor of 24.
+    """
+    if 24 % hours != 0:
+        raise ValueError('The number of hours must be a divisor of 24')
+    hours = range(0, 24, hours)
+    select_hours = iris.Constraint(
+        time=lambda cell: cell.point.hour in hours)
+    return cube.extract(select_hours)
+
+
+def resample_time(cube, month=None, day=None, hour=None):
```

Comment:
This is slightly confusing, how is this function different from `resample_hours`?

Does `resample_time(cube, hour=6)` give the same out put as `resample_hours(cube, hours=6)`

Reply:
No, 

resample_time(cube, hour=6) -> Daily values taken at 6:00
resample_hour(cube, every_n_hours=6) -> Six hourly values at 0:00, 6:00, 12:00, 18:00

Reply:
Nice, I suggest adding these examples to the docstring to avoid confusion to make the intent clear.